Adequacy patch - CSR

hugo-antoine-rtei opened the pull request:
Generating an executable for beta 4 of adq patch

Milos-RTEi committed:
2-8. replace seeds with two thresholds

Milos-RTEi committed:
9. resetAdqPatchParameters

Milos-RTEi committed:
10. re-name lmrViolations.h

Milos-RTEi committed:
25-26. re-name AdequacyPatchConfigure to AdequacyPatchOptions

Milos-RTEi committed:
27. cellValue change

Milos-RTEi committed:
28. is already implemented + small refacto

Milos-RTEi committed:
small refacto

Milos-RTEi committed:
29. updateButton function

Commit:
13. renaming

Commit:
14. refactoring

Commit:
15. remove redundancy

Commit:
16. refactoring

Milos-RTEi committed:
20. separate LHS to functions

Milos-RTEi committed:
20. separate RHS to functions

Milos-RTEi committed:
clean

Milos-RTEi committed:
23. separate Costs into functions

Milos-RTEi committed:
22. separate Bounds into functions

Milos-RTEi committed:
21. separate Construction of variables into functions

Milos-RTEi committed:
19. separate quad-solver into sub-functions

Milos-RTEi committed:
11. separate getHoursRequiringCurtailmentSharing into two sub-functions

Milos-RTEi committed:
17+18+24. make OPT_ functions private methods and make public run method

Milos-RTEi committed:
clean

Commit:
clean

Milos-RTEi committed:
clean

Milos-RTEi committed:
Merge pull request #48 from rte-i/feature/rte-comments-csr-batch1

RTE comments on CSR

Milos-RTEi committed:
Merge branch 'develop' into feature/adq-patch-3

hugo-antoine-rtei committed:
Merge pull request #715 from rte-i/feature/adq-patch-3

Feature/adq patch 3 - attempt to fix windows CI pipeline

Milos-RTEi committed:
Merge branch 'develop' into feature/adq-patch

Milos-RTEi committed:
0 as a default value, and (value > threshold)

Milos-RTEi committed:
4. add else if for AdqPatchPTO::isDens

Milos-RTEi committed:
bring back some debugg logs for easy result check

Milos-RTEi committed:
2. make calculateCsrParameters method of HOURLY_CSR_PROBLEM

Milos-RTEi committed:
clean

Milos-RTEi committed:
1. use std::vector for sumENS

Milos-RTEi committed:
9. repack CSR ui input data in parameters.h

Milos-RTEi committed:
8. placing all functions inside adequacy_patch.h to namespace Adequcy patch

Milos-RTEi committed:
7. re-pack CSR run into weeklyOptProblem->solve method

Milos-RTEi committed:
adopt RTE syntax for switch/case

Milos-RTEi committed:
remove comments

Milos-RTEi committed:
Merge pull request #54 from rte-i/develop

Merge latest rte/develop to rtei/feature/adq-patch

Milos-RTEi committed:
change RC number

Milos-RTEi committed:
Merge pull request #55 from rte-i/change-rc-num

change RC number

Milos-RTEi committed:
change beta release number

Milos-RTEi committed:
Merge pull request #56 from rte-i/change-rc-num

change beta release number

Milos-RTEi committed:
unit test for CSR-calculateAreaFlowBalance

Milos-RTEi committed:
small refacto

Milos-RTEi committed:
unit test bug fix

Milos-RTEi committed:
Merge pull request #52 from rte-i/feature/rte-csr-comments-batch2

RTE comments implementation [1-4]+[6-9]

Commit:
4. remove unused

Commit:
6.1 MemAllocMemset

Commit:
5. remove return

Commit:
3. remove debug log

Commit:
2. move to csr

Commit:
1. refacto

Commit:
6.2 hide column

Commit:
6.2

Commit:
3. move log

Commit:
3. move log..

Commit:
clean

flomnes committed:
Merge remote-tracking branch 'rte/develop' into feature/issue57

Milos-RTEi committed:
Merge pull request #59 from AntaresSimulatorTeam/feature/issue57

Fix conflicts with rte/develop

Milos-RTEi committed:
Merge pull request #58 from rte-i/feature/issue57

Feature/issue57

Milos-RTEi committed:
small fix year count

Milos-RTEi committed:
hot start

Milos-RTEi committed:
relax var bounds

Milos-RTEi committed:
remove ENS <= densNew constraint, set as var bound

Milos-RTEi committed:
remove set values to NaN if csr solver fails

Milos-RTEi committed:
relax bining constrains for csr solver

Milos-RTEi committed:
make csr crash log more expressive

Milos-RTEi committed:
clean some logs

Milos-RTEi committed:
beta 3 -> beta 4

Milos-RTEi committed:
Merge pull request #61 from rte-i/feature/fix-solver-issues

Feature/fix solver issues

hugo-antoine-rtei committed:
Resolved merge conflicts by keeping rtei's changes

Milos-RTEi committed:
wrong relaxation of bc - bug fix

Milos-RTEi committed:
Merge pull request #62 from rte-i/feature/csr-quad-solver-relaxation-bug-fix

wrong relaxation of bc - bug fix

hugo-antoine-rtei committed:
Merge pull request #786 from rte-i/feature/adq-patch

Beta 5 executable

Milos-RTEi committed:
logg out triggered csr hours

Milos-RTEi committed:
refacto 1e-3

Milos-RTEi committed:
logg out csr warnings - refacto

Milos-RTEi committed:
update antares output dens value

Milos-RTEi committed:
more expressive log

Milos-RTEi committed:
update dens and use it as pto

Milos-RTEi committed:
dont include linear cost coeff - temporary beta6 only

Milos-RTEi committed:
beta5->beta6

Milos-RTEi committed:
Merge pull request #63 from rte-i/feature/csr-solver-issues-all-zeros

client reports all zeros for flows variables for csr hours

Milos-RTEi committed:
Merge pull request #795 from rte-i/feature/adq-patch

Make executables!

flomnes commented on src/solver/optimisation/opt_appel_solveur_quadratique.cpp:
```diff
@@ -183,3 +183,153 @@ bool OPT_AppelDuSolveurQuadratique(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudr
     }
 }
+
+void setInteriorPointProblem(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre,
+                             PROBLEME_POINT_INTERIEUR& Probleme)
+{
+    double ToleranceSurLAdmissibilite;
+    int ChoixToleranceParDefautSurLAdmissibilite;
+    double ToleranceSurLaStationnarite;
+    int ChoixToleranceParDefautSurLaStationnarite;
```
Move all adq-patch stuff to **adequacy_patch_curtailment_sharing_problem.cpp**.

Create a directory called adequacy_patch_csr/

flomnes commented on src/solver/optimisation/opt_appel_solveur_quadratique.cpp:
```diff
@@ -183,3 +183,153 @@ bool OPT_AppelDuSolveurQuadratique(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudr
+    double ToleranceSurLaComplementarite;
+    int ChoixToleranceParDefautSurLaComplementarite;
+
+    ToleranceSurLAdmissibilite = 1.e-5;
+    ChoixToleranceParDefautSurLAdmissibilite = OUI_PI;
+
+    ToleranceSurLaStationnarite = 1.e-5;
+    ChoixToleranceParDefautSurLaStationnarite = OUI_PI;
+
+    ToleranceSurLaComplementarite = 1.e-5;
+    ChoixToleranceParDefautSurLaComplementarite = OUI_PI;
```
You can remove those default value

flomnes commented on src/solver/optimisation/opt_appel_solveur_quadratique.cpp:
```diff
@@ -183,3 +183,153 @@ bool OPT_AppelDuSolveurQuadratique(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudr
+    Probleme.NombreDeVariables = ProblemeAResoudre->NombreDeVariables;
+    Probleme.TypeDeVariable = ProblemeAResoudre->TypeDeVariable;
+
+    Probleme.VariableBinaire = (char*)ProblemeAResoudre->CoutsReduits;
```
double* -> char* casting, avoid this. You could create binary variables by accident.

double -> char. In most cases, the char value will be equal to PI_OUI=1. Allocate a real array and set all values to `PI_NON`.

flomnes commented on src/solver/optimisation/opt_appel_solveur_quadratique.cpp:
```diff
@@ -183,3 +183,153 @@ bool OPT_AppelDuSolveurQuadratique(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudr
+
+#ifndef NDEBUG
+
+    {
+        logs.info();
+
+        logs.info() << LOG_UI_DISPLAY_MESSAGES_OFF;
+
+        logs.info() << "Here is the trace:";
+        for (Var = 0; Var < ProblemeAResoudre->NombreDeVariables; Var++)
+        {
+            logs.info().appendFormat("Variable %ld cout lineaire %e  cout quadratique %e",
+                                     Var,
+                                     ProblemeAResoudre->CoutLineaire[Var],
+                                     ProblemeAResoudre->CoutQuadratique[Var]);
+        }
+        for (int Cnt = 0; Cnt < ProblemeAResoudre->NombreDeContraintes; Cnt++)
+        {
+            logs.info().appendFormat("Constraint %ld sens %c B %e",
+                                     Cnt,
+                                     ProblemeAResoudre->Sens[Cnt],
+                                     ProblemeAResoudre->SecondMembre[Cnt]);
+
+            int il = ProblemeAResoudre->IndicesDebutDeLigne[Cnt];
+            int ilMax = il + ProblemeAResoudre->NombreDeTermesDesLignes[Cnt];
+            for (; il < ilMax; ++il)
+            {
+                Var = ProblemeAResoudre->IndicesColonnes[il];
+                logs.info().appendFormat(
+                  "      coeff %e var %ld xmin %e xmax %e type %ld",
+                  ProblemeAResoudre->CoefficientsDeLaMatriceDesContraintes[il],
+                  Var,
+                  ProblemeAResoudre->Xmin[Var],
+                  ProblemeAResoudre->Xmax[Var],
+                  ProblemeAResoudre->TypeDeVariable[Var]);
+            }
+        }
+    }
```
Plug this into a function `debug_blah`

flomnes commented on src/solver/optimisation/opt_appel_solveur_quadratique.cpp:
```diff
@@ -183,3 +183,153 @@ bool OPT_AppelDuSolveurQuadratique(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudr
+#endif
+}
+
+bool OPT_AppelDuSolveurQuadratique_CSR(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre,
```
ADQ_PATCH_CSR

flomnes commented on src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_quadratique.cpp:
```diff
@@ -91,3 +93,331 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeQuadratique(PROBLEME_HEBDO*
     MemFree(Colonne);
 }
+
+// void setConstraintsOnENS(PROBLEME_HEBDO* ProblemeHebdo,
+//                          HOURLY_CSR_PROBLEM& hourlyCsrProblem,
+//                          double* Pi,
+//                          int* Colonne)
+// {
+//     int hour = hourlyCsrProblem.hourInWeekTriggeredCsr;
+//     int Area;
+//     int Var;
+//     int NombreDeTermes;
+//     CORRESPONDANCES_DES_VARIABLES* CorrespondanceVarNativesVarOptim;
+//     PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre;
+//     std::string NomDeLaContrainte;
+
+//     ProblemeAResoudre = ProblemeHebdo->ProblemeAResoudre;
+//     CorrespondanceVarNativesVarOptim = ProblemeHebdo->CorrespondanceVarNativesVarOptim[hour];
+
+//     // constraint: ENS < DENS_new
+//     for (Area = 0; Area < ProblemeHebdo->NombreDePays; ++Area)
+//     {
+//         if (ProblemeHebdo->adequacyPatchRuntimeData.areaMode[Area]
+//             == Data::AdequacyPatch::physicalAreaInsideAdqPatch)
```
Remove this commented code

flomnes commented on src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_quadratique.cpp:
```diff
@@ -91,3 +93,331 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeQuadratique(PROBLEME_HEBDO*
+            && ProblemeHebdo->adequacyPatchRuntimeData.extremityAreaMode[Interco]
+                 == Antares::Data::AdequacyPatch::physicalAreaInsideAdqPatch)
+        {
+            TransportCost = ProblemeHebdo->CoutDeTransport[Interco];
```
unused

flomnes commented on src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_quadratique.cpp:
```diff
@@ -91,3 +93,331 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeQuadratique(PROBLEME_HEBDO*
+            hourlyCsrProblem.numberOfConstraintCsrAreaBalance[Area]
+              = ProblemeAResoudre->NombreDeContraintes;
+
+            NomDeLaContrainte = "Area Balance, Area:" + std::to_string(Area) + "; "
+                                + ProblemeHebdo->NomsDesPays[Area];
+            ;
```
""

flomnes commented on src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_quadratique.cpp:
```diff
@@ -91,3 +93,331 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeQuadratique(PROBLEME_HEBDO*
+                        Colonne[NombreDeTermes] = Var;
+                        NombreDeTermes++;
+
+                        // logs.debug()
+                        //   << "Interco:" + std::to_string(Interco) << ". Between:["
+                        //   << ProblemeHebdo
+                        //        ->NomsDesPays[ProblemeHebdo->PaysOrigineDeLInterconnexion[Interco]]
+                        //   << "]-["
+                        //   << ProblemeHebdo
+                        //        ->NomsDesPays[ProblemeHebdo->PaysExtremiteDeLInterconnexion[Interco]]
+                        //   << "], with Poids(coeff):" + std::to_string(Poids) + " inserted to LHS!";
```
Remove comments

flomnes commented on src/solver/optimisation/opt_construction_variables_optimisees_quadratique.cpp:
```diff
@@ -52,8 +52,135 @@ void OPT_ConstruireLaListeDesVariablesOptimiseesDuProblemeQuadratique(PROBLEME_H
         CorrespondanceVarNativesVarOptim->NumeroDeVariableDeLInterconnexion[Interco]
           = NombreDeVariables;
-        ProblemeAResoudre->TypeDeVariable[NombreDeVariables] = VARIABLE_BORNEE_DES_DEUX_COTES;
+        ProblemeAResoudre->TypeDeVariable[NombreDeVariables] = VARIABLE_BORNEE_INFERIEUREMENT;
```
If you don't call this function, there is no need to change it, right ?

flomnes commented on src/solver/optimisation/opt_construction_variables_optimisees_quadratique.cpp:
```diff
@@ -52,8 +52,135 @@ void OPT_ConstruireLaListeDesVariablesOptimiseesDuProblemeQuadratique(PROBLEME_H
+    ProblemeAResoudre = ProblemeHebdo->ProblemeAResoudre;
+    CORRESPONDANCES_DES_VARIABLES* CorrespondanceVarNativesVarOptim;
+    CorrespondanceVarNativesVarOptim = ProblemeHebdo->CorrespondanceVarNativesVarOptim[hour];
+    int NumberOfVariables = ProblemeAResoudre->NombreDeVariables;
```
```suggestion
    int& NumberOfVariables = ProblemeAResoudre->NombreDeVariables;
```
Then there is no need to copy it back to `ProblemeAResoudre` at the end.

Milos-RTEi committed:
10. bring back type of var to origin quad solver - mistake

Milos-RTEi committed:
2. remove default values for the solver accuracy

Milos-RTEi committed:
change function  name to ADQ_PATCH_CSR

Milos-RTEi committed:
6. remove comments

Milos-RTEi committed:
7. remove nused transport cost

Milos-RTEi committed:
9. remove comments

Milos-RTEi committed:
11. refacto - int NumberOfVariables

Milos-RTEi committed:
8. remove NomDeLaContrainte

Milos-RTEi committed:
bring back linear hurdle costs

Milos-RTEi committed:
set to zero ENS & spillage if below threshold (avoid LOLP=100% dummy values)

Milos-RTEi committed:
small refacto

Milos-RTEi committed:
4. make subfunction CSR_DEBUG_BLAH

Milos-RTEi committed:
1. move csr problem functions to separate directory

Milos-RTEi committed:
12. Add "Include hurdle cost" false/true button

Milos-RTEi committed:
Merge pull request #798 from rte-i/feature/issue64-milos

Make executables beta official

Milos-RTEi committed:
refacto avoid bugs

Milos-RTEi committed:
refacto avoid bugs

Milos-RTEi committed:
refacto bugs

Milos-RTEi committed:
Merge pull request #806 from rte-i/feature/issue64-milos

Create executables newest development

flomnes committed:
Merge 'rte/develop' into feature/do-not-merge-csr-generate-executables

flomnes committed:
Fix build

Rename LinkCapacityForAdequacyPatchFirstStep -> ntcSetToZeroStatus_AdqPatchStep1

Milos-RTEi committed:
Merge pull request #65 from rte-i/feature/issue64-milos

Issue 64

flomnes commented on src/solver/optimisation/adequacy_patch_csr/construct_problem_variables.cpp:
```diff
@@ -0,0 +1,163 @@
+            NumberOfVariables++;
+        }
+    }
+    ProblemeAResoudre->NombreDeVariables = NumberOfVariables;
```
Make `NumberOfVariables` a reference to `ProblemeAResoudre->NombreDeVariables`, same as below

flomnes commented on src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp:
```diff
@@ -0,0 +1,203 @@
+{
+    PROBLEME_POINT_INTERIEUR Probleme;
+    setInteriorPointProblem(ProblemeAResoudre, Probleme);
+    PI_Quamin(&Probleme); // resolution
```
Slightly clearer version
```suggestion
    std::unique_ptr<PROBLEME_POINT_INTERIEUR> Probleme = buildInteriorPointProblem(ProblemeAResoudre);
    PI_Quamin(Probleme.get()); // resolution
```

flomnes commented on src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp:
```diff
@@ -0,0 +1,203 @@
+    setInteriorPointProblem(ProblemeAResoudre, Probleme);
+    PI_Quamin(&Probleme); // resolution
+    ProblemeAResoudre->ExistenceDUneSolution = Probleme.ExistenceDUneSolution;
+    if (ProblemeAResoudre->ExistenceDUneSolution == OUI_PI)
```
Why do you need this copy ? Can you test `Probleme.ExistenceDUneSolution` directly ?

flomnes commented on src/solver/simulation/economy.cpp:
```diff
@@ -93,11 +97,23 @@ void AdequacyPatchOptimization::solve(Variable::State& state, int hourInTheYear,
     ::SIM_RenseignementProblemeHebdo(*problemeHebdo, state, numSpace, hourInTheYear);
     OPT_OptimisationHebdomadaire(problemeHebdo, numSpace);
+
+    std::set<int> hoursRequiringCurtailmentSharing = getHoursRequiringCurtailmentSharing(numSpace);
```
```suggestion
    const std::set<int> hoursRequiringCurtailmentSharing = getHoursRequiringCurtailmentSharing(numSpace);
```

flomnes replied:
I suggest that you place lines 101-111 into a dedicated function.

flomnes commented on src/solver/simulation/economy.cpp:
```diff
@@ -173,6 +189,45 @@ bool Economy::simulationBegin()
+    std::set<int> triggerCsrSet;
+    for (int i = 0; i < nbHoursInAWeek; ++i)
+    {
+        if ((int)sumENS[i] > threshold)
```
Why do you cast `sumENS[i]` to `int` ?

flomnes commented on src/solver/simulation/economy.cpp:
```diff
@@ -173,6 +189,45 @@ bool Economy::simulationBegin()
+                                                                            uint numSpace)
+{
+    float threshold
+      = pProblemesHebdo[numSpace]->adqPatchParams->ThresholdInitiateCurtailmentSharingRule;
```
Use a `double` for consistency ?

flomnes commented on src/solver/simulation/economy.h:
```diff
@@ -46,7 +46,7 @@ class EconomyWeeklyOptimization
 public:
     using Ptr = std::unique_ptr<EconomyWeeklyOptimization>;
-    virtual void solve(Variable::State& state, int hourInTheYear, uint numSpace) = 0;
+    virtual void solve(Variable::State& state, int hourInTheYear, uint numSpace, uint w) = 0;
```
```suggestion
    virtual void solve(Variable::State& state, int hourInTheYear, uint numSpace, uint week) = 0;
```

flomnes commented on src/solver/simulation/economy.h:
```diff
@@ -58,14 +58,18 @@ class AdequacyPatchOptimization : public EconomyWeeklyOptimization
+    void solve(Variable::State& state, int hourInTheYear, uint numSpace, uint w) override;
+
+    vector<double> calculateENSoverAllAreasForEachHour(uint numSpace);
+    std::set<int> identifyHoursForCurtailmentSharing(vector<double> sumENS, uint numSpace);
```
Make these 2 methods `private`, they're implementation details.

flomnes commented on src/solver/simulation/sim_structure_probleme_economique.h:
```diff
@@ -495,8 +498,15 @@ struct AdequacyPatchParameters
 };
 
+class HOURLY_CSR_PROBLEM;
+
```
Why do you need a forward declaration here ?

flomnes commented on src/solver/simulation/sim_structure_probleme_economique.h:
```diff
@@ -707,4 +717,39 @@ struct PROBLEME_HEBDO
+    void setProblemCost();
+    void solveProblem(uint week, int year);
+public:
+    void run(uint week, int year);
```
Good ! Implementation details are `private`, the API is really simple.

flomnes commented on src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_RHS.cpp:
```diff
@@ -0,0 +1,220 @@
+              = MatriceDesContraintesCouplantes->SecondMembreDeLaContrainteCouplante[hour];
+            // logs.debug() << Cnt << ": Hourly bc: Existing-RHS[" << Cnt
+            //              << "] = " << SecondMembre[Cnt] << " (CntCouplante = " << CntCouplante
+            //              << ")";
```
```suggestion
```

flomnes commented on src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_RHS.cpp:
```diff
@@ -0,0 +1,220 @@
+/*
+** Copyright 2007-2018 RTE
```
```suggestion
** Copyright 2007-2022 RTE
```

flomnes commented on src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_RHS.cpp:
```diff
@@ -0,0 +1,220 @@
+/*
+** Copyright 2007-2018 RTE
+** Authors: Antares_Simulator Team
```
```suggestion
** Authors: RTE-international / Redstork / Antares_Simulator Team
```

flomnes commented on src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_RHS.cpp:
```diff
@@ -0,0 +1,220 @@
+                    //   << ProblemeHebdo
+                    //        ->NomsDesPays[ProblemeHebdo->PaysExtremiteDeLInterconnexion[Interco]]
+                    //   << "]"
+                    //   << ". ValueOfFlow: " << ValueOfFlow << ". Poids: " << Poids;
```
```suggestion
```

flomnes commented on src/solver/optimisation/opt_gestion_second_membre_cas_quadratique.cpp:
```diff
@@ -25,6 +25,8 @@
 */
 
+const double csrSolverRelaxationRHS = 1e-3;
+
```
Double definition. This one is unused, so I'd remove it.

flomnes commented on src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_RHS.cpp:
```diff
@@ -0,0 +1,220 @@
+                //              << "] = " << SecondMembre[Cnt] << " (CntCouplante = " << CntCouplante
+                //              << ")"
+                //              << ". Area:" << Area << ", Palier:" << Palier << ", Poids" << Poids
+                //              << ", ValueOfVar:" << ValueOfVar;
```
```suggestion
```

flomnes commented on src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_RHS.cpp:
```diff
@@ -0,0 +1,220 @@
+                //              << ". Area:" << Area << ", Palier:" << Palier << ", Poids" << Poids
+                //              << ", ValueOfVar:" << ValueOfVar;
+            }
+            if (MatriceDesContraintesCouplantes->SensDeLaContrainteCouplante == '<')
+            {
+                SecondMembre[Cnt] += csrSolverRelaxationRHS;
+            }
+            else if (MatriceDesContraintesCouplantes->SensDeLaContrainteCouplante == '>')
+            {
+                SecondMembre[Cnt] -= csrSolverRelaxationRHS;
+            }
```
What happens to ranged binding constraints ? Equality binding constraints ?

flomnes commented on src/libs/antares/study/fwd.h:
```diff
@@ -592,6 +592,23 @@ enum LinkCapacityForAdequacyPatchFirstStep
+const float adqPatchDefaultValueThresholdInitiateCurtailmentSharingRule = 0.0;
+//! A default threshold value for display local matching rule violations
+const float adqPatchDefaultValueThresholdDisplayLocalMatchingRuleViolations = 0.0;
+
```
`double` ? Also, I suggest you move this somewhere else, `fwd.h` isn't meant for this.

flomnes commented on src/libs/antares/study/parameters.cpp:
```diff
@@ -201,12 +233,24 @@ void Parameters::resetSeeds()
     adqPatch.localMatching.setToZeroOutsideInsideLinks = true;
     adqPatch.localMatching.setToZeroOutsideOutsideLinks = true;
+    adqPatch.curtailmentSharing.priceTakingOrder = Data::AdequacyPatch::AdqPatchPTO::isDens;
+    adqPatch.saveIntermediateResults = false;
```
This seems to be completely unused, remove ? This remark applies to variables in the problem, `Parameters`, and buttons in the GUI.

flomnes replied:
Forget it

flomnes commented on src/solver/optimisation/adequacy_patch.h:
```diff
@@ -37,79 +37,86 @@ namespace Data
+** ** \param num An integer
+** ** \return
+** */
+void addArray(std::vector<double>& A, double* B, int num);
```
You may use `A.size()`, instead of passing `num`

flomnes committed:
Merge pull request #811 from AntaresSimulatorTeam/feature/csr-merge-develop

Merge develop branch to solve conflicts

Milos-RTEi committed:
Merge remote-tracking branch 'upstream/feature/csr-merge-develop' into feature/keep-track-with-rte-develop

Milos-RTEi committed:
1. Make NumberOfVariables a reference var

Milos-RTEi committed:
change function name CSR_DEBUG_

Milos-RTEi committed:
3. test Probleme.ExistenceDUneSolution directly

Milos-RTEi committed:
2. Slightly clearer version

Milos-RTEi committed:
4. make hoursRequiringCurtailmentSharing const var

Milos-RTEi committed:
5. avoid cast sumENS[i] to int!

Milos-RTEi committed:
17. move default thresholds to adequacy_patch.h

Milos-RTEi committed:
20. use A.size() in addArray function

Milos-RTEi committed:
19. create dedicated method solveCSR

Milos-RTEi committed:
7. name change

Milos-RTEi committed:
8. make methods of AdequacyPatchOptimization private

Milos-RTEi committed:
11. 14. 16 delete unused logs

Milos-RTEi committed:
9.  delete forward declaration of HOURLY_CSR_PROBLEM

Milos-RTEi committed:
15. remove Double definition.

Milos-RTEi committed:
12. 13 update copyright & Authors for adq-patch files

Milos-RTEi committed:
Merge pull request #68 from rte-i/feature/issue66-milos

Issue-66 comment implementation

Milos-RTEi committed:
Merge pull request #67 from rte-i/feature/keep-track-with-rte-develop

KEEP TRACK WITH RTE DEVELOP

Milos-RTEi committed:
Merge remote-tracking branch 'origin/develop' into feature/adq-patch-4

Milos-RTEi committed:
make solveCSR private method

Milos-RTEi committed:
use auto instead of std::unique_ptr<PROBLEME_POINT_INTERIEUR>

Milos-RTEi committed:
float->double

Milos-RTEi committed:
remove saveIntermediateResults

Milos-RTEi committed:
Merge pull request #70 from rte-i/feature/issue66-update

CSR comments implementation update

Milos-RTEi committed:
Merge pull request #69 from rte-i/feature/issue66-update

Issue-66 update

flomnes committed:
Merge pull request #825 from rte-i/feature/issue66-milos

Implement latest CSR comments

Milos-RTEi committed:
code smells parameters.cpp

Milos-RTEi committed:
critical-CodeSmell-adequacy_patch.cpp

Milos-RTEi committed:
critical-codeSmells- set_problem_cost.cpp

Milos-RTEi committed:
info-CodeSmell - remove todo-s

Milos-RTEi committed:
CodeSmells - global var should be const

Milos-RTEi committed:
NULL->nullptr

Milos-RTEi committed:
use default destructor

Milos-RTEi committed:
protected->private in adq-patch-area-grid.h

Milos-RTEi committed:
use default constructor in adequacy-patch-areas

Milos-RTEi committed:
Add the "explicit" keyword to AdequacyPatchAreas constructor

Milos-RTEi committed:
NULL->nullptr in adequacy-patch-options.cpp

Milos-RTEi committed:
= default, AdequacyPatchOptions

Milos-RTEi committed:
Add the "explicit" keyword to AdequacyPatchOptions constructor

Milos-RTEi committed:
Remove these useless parentheses

Milos-RTEi committed:
Remove these useless parentheses

Milos-RTEi committed:
Remove the unused parameter "weekNb",

Milos-RTEi committed:
Move these 4 #include directives to the top of the file

Milos-RTEi committed:
declaration shadows a local variable

Milos-RTEi committed:
Value stored to 'ProblemeAResoudre' is never read

Milos-RTEi committed:
function should be declared "const"

Milos-RTEi committed:
Use "std::make_unique" to construct "std::unique_ptr

Milos-RTEi committed:
Value stored to 'out' during its initialization is never read

Milos-RTEi committed:
major-CodeSmells-adequacy-patch-options.cpp

Milos-RTEi committed:
remove "never read" variables

Milos-RTEi committed:
const& buttonType in updateButton

Milos-RTEi committed:
Replace this "enum" with "enum class"  AdqPatchPTO

Milos-RTEi committed:
double*->const double* in addArray

Milos-RTEi committed:
MinorCodeSmell-s construct_LHS

Milos-RTEi committed:
MinorCodeSmell-s construct_RHS

Milos-RTEi committed:
MinorCodeSmell-s construct_variables

Milos-RTEi committed:
MinorCodeSmell-s cost_function

Milos-RTEi committed:
MinorCodeSmell-s set_var_bounds

Milos-RTEi committed:
MinorCodeSmell-s solve_problem

Milos-RTEi committed:
MinorCodeSmell-s economy.cpp

Milos-RTEi committed:
Remove redundant public  access specifier

Milos-RTEi committed:
Remove this redundant access specifier

Milos-RTEi committed:
MinorCodeSmell-s adq-patch-unit-tests

Milos-RTEi committed:
Concatenate  namespace with the nested ones.

Milos-RTEi committed:
Use "override" or "final" instead of "virtual".

Milos-RTEi committed:
Remove public redundant access specifier;

Milos-RTEi committed:
MinorCodeSmell-s adequacy-patch-areas.h

Milos-RTEi committed:
Concatenate namespace with the nested ones.

Milos-RTEi committed:
Make the type of auto& study ; a reference-to-const

Milos-RTEi committed:
MinorCodeSmell-s AdequacyPatchOptions

Milos-RTEi committed:
"not"-> "!" , "and"-> "&&"

Milos-RTEi committed:
Merge pull request #837 from rte-i/feature/code-smells-csr

Check how many remaining code smells in Sonar Cloud

flomnes committed:
Add adequacy-patch CSR study format changes (#817)

Milos-RTEi commented:
Hey @flomnes. I failed SonarCloud execution 2times today after I merged #837.
I thought I have some access/verification issue on my side..but now I see it's failing again when you merged #817. 
How did I mess up SonarCloud with my code-smells-implementation?

flomnes commented:
> Hey @flomnes. I failed SonarCloud execution 2times today after I merged #837. I thought I have some access/verification issue on my side..but now I see it's failing again when you merged #817. How did I mess up SonarCloud with my code-smells-implementation?

It's our fault. We're working to restore this feature.

Milos-RTEi committed:
beta 7

Milos-RTEi committed:
create GUI for csr var relax button

Milos-RTEi committed:
connect GUI to logic

Milos-RTEi committed:
threshold e-3 >e-5

Milos-RTEi committed:
Merge pull request #842 from rte-i/feature/request-from-elia-for-csr-relax-gui

Beta 7 Executables for Elia

Milos-RTEi committed:
fix (GUI wont pass smaller than 10^-6)

Milos-RTEi committed:
Merge pull request #846 from rte-i/feature/request-from-elia-for-csr-relax-gui

Fix for beta7

Milos-RTEi committed:
Value stored to 'PaliersThermiquesDuPays' is never read

Milos-RTEi committed:
"std::string" -> "std::string_view"

Milos-RTEi committed:
make getHoursRequiringCurtailmentSharing const method

Milos-RTEi committed:
remove unused var

Milos-RTEi committed:
assign ProblemeAResoudre properly

Milos-RTEi committed:
Merge pull request #847 from rte-i/feature/code-smells-csr

Implementation of some remaining code smells that can be fixed in CSR branch

Milos-RTEi committed:
Merge pull request #71 from rte-i/feature/code-smells-csr

Fix some of the code smells in CSR solution

Milos-RTEi committed:
Merge pull request #77 from rte-i/feature/request-from-elia-for-csr-relax-gui

Feature/request from elia for csr relax gui

Milos-RTEi committed:
Reporting of DENS_new in GUI result for all hours

Milos-RTEi committed:
Check CSR cost-fun-value prior&after CSR-opt - GUI

Milos-RTEi committed:
refacto due to pre-calculating dens_new for all hours prior to csr optim

Milos-RTEi committed:
add "SpilledenergyAfterCSR" column

Milos-RTEi committed:
TODO: densNewValues

Milos-RTEi committed:
Correction of MRG.PRICE Price Cap in hours LOLE>0.5 after CSR

Milos-RTEi committed:
refacto "DENS_new" parameter calc, same thing at the end!?

Milos-RTEi committed:
add DTG. MRG INT/FIN columns to GUI

Milos-RTEi committed:
copy-paste spilled Energy values into spilled Energy values after CSR prior to CSR

Milos-RTEi committed:
Check CSR cost value prior to and after - UNTESTED

Milos-RTEi committed:
refacto on DTG MRG CSR columns GUI

Milos-RTEi committed:
change log to info in "Check CSR cost value"

Milos-RTEi committed:
test implementation csr fun cost prior/after

Milos-RTEi committed:
refacto LMR violations according to the agreement with Daniel + clean

Milos-RTEi committed:
csr-postProcess - adjust ENS value and calc dtgMrgCsr

Milos-RTEi committed:
small refacto

Milos-RTEi committed:
refacto - adjust mrgPrice after caclulating dtgMrgCsr and new Ens

Milos-RTEi committed:
log update

Milos-RTEi committed:
refacto setToZeroIfBelowThreshold

Milos-RTEi committed:
update/refacto log csr-solver

Milos-RTEi committed:
refacto logs @ solve_problem.cpp

Milos-RTEi committed:
typo

Milos-RTEi committed:
beta7 -> beta8

Milos-RTEi committed:
Merge pull request #931 from rte-i/feature/adq-patch-sequel

Adequacy Patch Sequel

Milos-RTEi committed:
add prefix [adq-patch] to info logs

Milos-RTEi committed:
Merge pull request #933 from rte-i/feature/adq-patch-sequel

Adequacy Patch Sequel

Milos-RTEi committed:
move adqPatchPostProcess after HydroRemix

Milos-RTEi committed:
refactor + change -  storeOrDisregardInteriorPointResults

Milos-RTEi committed:
beta 9.1

Milos-RTEi committed:
Merge pull request #953 from rte-i/feature/adq-patch-sequel

Adequacy Patch CSR Solution Update

flomnes commented on docs/reference-guide/13-file-format.md:
```diff
@@ -1,6 +1,19 @@
+Add variable `LMR VIOL.` for each area. Thematic trimming allows to enable/disable this output variable, see section `variables selection` in file **settings/generaldata.ini**.
+
+### Output
+A variable `LMR VIOL.` is added for each area if `include-adq-patch` is set to `true`.
```
Incomplete, those are missing
- SPIL. ENRG. CSR
- DTG MRG CSR

Milos-RTEi committed:
refactor solveCSR

Milos-RTEi committed:
add "- dtgMrg" to densNew formula

Milos-RTEi committed:
beta 10

Milos-RTEi committed:
Merge pull request #955 from rte-i/feature/adq-patch-sequel

Adequacy-patch Update - beta 10 version

Milos-RTEi committed:
Merge branch 'develop' into feature/adq-patch-sequel

Milos-RTEi committed:
fix error

Milos-RTEi committed:
update 13-file-format.md for adq-patch CSR

Milos-RTEi committed:
update 14-adequacy-patch.md for adq-patch CSR

Milos-RTEi committed:
change CSR objective function (Elia)

Milos-RTEi committed:
change CSR objective function (Elia) - doc update

Milos-RTEi committed:
beta11

Milos-RTEi committed:
change CSR objective function (Elia) - doc update

Milos-RTEi committed:
Merge branch 'feature/do-not-merge-csr-generate-executables' into feature/adq-patch-sequel

Milos-RTEi committed:
Merge pull request #991 from rte-i/feature/adq-patch-sequel

Adequacy-patch - beta 11 version

Milos-RTEi committed:
14-adq-patch-doc-bug fix (unrecognized characters)

Milos-RTEi committed:
Merge pull request #992 from rte-i/feature/adq-patch-sequel

Adequacy-patch - beta 11 version

Milos-RTEi committed:
14-adq-patch-doc-bug fix (unrecognized characters)-2

Milos-RTEi committed:
Merge pull request #993 from rte-i/feature/adq-patch-sequel

Adequacy-patch - beta 11 version

Milos-RTEi committed:
Merge branch 'feature/adq-patch-sequel' into develop

Milos-RTEi committed:
Merge pull request #81 from rte-i/develop

Merge latest Develop branch into adq-patch-sequel branch

Milos-RTEi committed:
Merge pull request #998 from rte-i/feature/adq-patch-sequel

Merge latest Develop branch into do-not-merge-csr-generate-executables

Milos-RTEi committed:
Fix [compile error]

Milos-RTEi committed:
Merge pull request #999 from rte-i/feature/adq-patch-sequel

Fix [compile error]
